src/formState.ts

```diff
@@ -373,10 +385,14 @@ export class FormState<
 export class ArrayFormState<
   V, T extends Validatable<any, V> = Validatable<any, V>
 > extends AbstractFormState<
-  T[], V[]
+  readonly T[], V[]
 > {
 
-  @observable.ref readonly $: T[]
+  @observable.ref protected fieldList: T[]
```

Comment:
这个为啥是 `protected` 的

Reply:
因为这个字段是定义在父类 `AbstractFormState` 上的，后者也确实会去消费它

Reply:
NIP：这个也应该是 `readonly` 吧？

Reply:
不是 readonly 的，比如 `ArrayFormState` 上的 `fieldList` 就是会被 `move` / `insert` 等操作修改的，不过这个字段不对外，所以 formstate-x 的使用方不会能够直接修改它的

Reply:
我看 `move/insert` 等操作似乎都在尽力避免修改 `fieldList` 的引用。可能我表述不清楚，我说的是：

 ```suggestion
  @observable.ref protected readonly fieldList: T[]
```

而不是

 ```suggestion
  @observable.ref protected fieldList:  readonly T[]
```

Reply:
嗯是可以 ` @observable.ref protected readonly fieldList: T[]` 的，不过之前没有通过 readonly 防 fromstate-x 自己代码的习惯，所以这里就不加了